perun-core/src/main/java/cz/metacentrum/perun/core/bl/SearcherBl.java

```diff
@@ -7,20 +7,50 @@
 import cz.metacentrum.perun.core.api.PerunSession;
 import cz.metacentrum.perun.core.api.Resource;
 import cz.metacentrum.perun.core.api.User;
+import cz.metacentrum.perun.core.api.Vo;
 import cz.metacentrum.perun.core.api.exceptions.AttributeNotExistsException;
 import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
 import cz.metacentrum.perun.core.api.exceptions.WrongAttributeAssignmentException;
 
 import java.time.LocalDate;
 import java.util.List;
 import java.util.Map;
+import java.util.stream.Collectors;
 
 /**
  * Searcher Class for searching objects by Map of Attributes
  *
  * @author Michal Stava <stavamichal@gmail.com>
  */
 public interface SearcherBl {
+	/**
+	 * Filter output from getGroups by vo_id.
+	 * @see #getGroups(PerunSession, Map)
+	 */
+	List<Group> getGroups(PerunSession sess, Vo vo, Map<String, String> attributesWithSearchingValues) throws InternalErrorException, AttributeNotExistsException, WrongAttributeAssignmentException;
+
+	/**
+	 * This method get Map of Attributes with searching values and try to find all groups, which have specific attributes in format.
+	 * Better information about format below. When there are more than 1 attribute in Map, it means all must be true "looking for all of them" (AND)
+	 *
+	 * IMPORTANT: can't get CORE ATTRIBUTES, it will skip any core attribute in map without information about it
+	 *
+	 * @param sess perun session
+	 * @param attributesWithSearchingValues map of attributes names
+	 *        when attribute is type String, so value is string and we are looking for total match (Partial is not supported now, will be supported later by symbol *)
+	 *        when attribute is type Integer, so value is integer in String and we are looking for total match
+	 *        when attribute is type List<String>, so value is String and we are looking for at least one total or partial matching element
+	 *        when attribute is type Map<String> so value is String in format "key=value" and we are looking total match of both or if is it "key" so we are looking for total match of key
+	 *        IMPORTANT: In map there is not allowed char '=' in key. First char '=' is delimiter in MAP item key=value!!!
+	 * @return list of users who have attributes with specific values (behavior above)
```

Comment:
``list of users`` -> ``list of groups``, also for the line below

Reply:
Copy&Paste mistake. Changed.